web3-android-mpc-provider/src/main/java/com/web3auth/web3_android_mpc_provider/EthereumTssAccount.java

```diff
@@ -72,6 +83,42 @@ public String signTransaction(BigInteger chainID, String toAddress, Double amoun
             txData = data;
         }
 
+        RawTransaction rawTransaction = RawTransaction.createTransaction(
+                chainID,
+                nonce,
+                gasLimit,
+                toAddress,
+                value,
+                txData
+        );
+
+        byte[] encodedTransaction = TransactionEncoder.encode(rawTransaction);
+        String encodedTransactionString = Base64.encodeToString(Hash.sha3(encodedTransaction), Base64.NO_WRAP);
+
+        Triple<BigInteger, BigInteger, Byte> signatureResult = sign(encodedTransactionString);
+
+        Byte v = signatureResult.getThird();
+        if (v < 27) {
+            v = (byte) ((chainID.byteValue() * 2) + (v + 35));
```

Comment:
This formula is for newer transactions, legacy uses the old one modification.
```suggestion
            v = (byte) v + 27;
```